src/main/java/com/indeed/operators/rabbitmq/api/RabbitMQApiClient.java

```diff
@@ -136,6 +155,24 @@ public void createOrUpdateShovel(final RabbitMQConnectionInfo connectionInfo, fi
         httpClient.newCall(req).execute();
     }
 
+    public void deleteShovel(final RabbitMQConnectionInfo connectionInfo, final String vhost, final String shovelName) throws IOException {
+        final String url = String.format(
+                "%s/parameters/%s/%s/%s",
+                buildRootUrl(connectionInfo),
+                "shovel",
+                encodeString(vhost),
+                encodeString(shovelName)
+        );
+
+        final Request req = new Request.Builder()
```

Comment:
This seems very formulaic - any way we can make this part of newCall() or otherwise reduce the duplication?

Reply:
Yes, this will all be replaced shortly.